src/test/resources/refaster/PreConditionsVerifier.java

```diff
@@ -0,0 +1,112 @@
+/*
+ * Copyright 2023 the original author or authors.
+ * <p>
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ * <p>
+ * https://www.apache.org/licenses/LICENSE-2.0
+ * <p>
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package foo;
+
+import java.util.List;
+import java.util.Map;
+
+import com.google.errorprone.refaster.annotation.AfterTemplate;
+import com.google.errorprone.refaster.annotation.BeforeTemplate;
+
+/**
+ * A refaster template to test when a `UsesType`and Preconditions.or should or should not be applied to the Preconditions check.
+ */
+public class PreConditionsVerifier {
+    public static class NoUsesTypeWhenBeforeTemplateContainsPrimitiveOrString {
```

Comment:
I choose as class name the intention of the recipe, like we use to do for the method name of a @Test method.. This make it both short and easy to grasp.